lxd/container_new.go

```diff
@@ -76,41 +79,79 @@ func getIps(c *lxc.Container) []shared.Ip {
 	return ips
 }
 
-func newStatus(c *lxc.Container, state lxc.State) shared.ContainerStatus {
-	status := shared.ContainerStatus{State: state.String(), StateCode: shared.State(int(state))}
-	if state == lxc.RUNNING {
-		status.Init = c.InitPid()
-		status.Ips = getIps(c)
-	}
-	return status
+type container interface {
+	RenderState() (*shared.ContainerState, error)
+	Reboot() error
+	Freeze() error
+	IsPrivileged() bool
+	IsRunning() bool
+	IsEmpheral() bool
+	IsSnapshot() bool
+	Shutdown(timeout time.Duration) error
+	Start() error
+	Stop() error
+	Unfreeze() error
+	Delete() error
+	CreateFromImage(hash string) error
+	Restore(sourceContainer container) error
+	Copy(source container) error
+	Rename(newName string) error
+	IDGet() int
+	NameGet() string
+	ArchitectureGet() int
+	PathGet(newName string) string
+	RootfsPathGet() string
+	TemplatesPathGet() string
+	StateDirGet() string
+	TemplateApply(trigger string) error
+	LogFilePathGet() string
+	LogPathGet() string
+	InitPidGet() (int, error)
+	IdmapSetGet() (*shared.IdmapSet, error)
+	ConfigReplace(newConfig containerLXDArgs) error
+	ConfigGet() containerLXDArgs
```

Comment:
Will sort them with the next patches but i want this accepted first, patches:

- Sort the interface
- Move all *lxdContainer code to lxd/container.go and sort the methods
- Rename *lxdContainer to *containerLXD
- Rename newLxdContainer to containerLXDLoad

About the *Get methods, with the Interface we don't have accessible Variables and i don't know if that's possible. This way we get clean structured accesses to the "object".

There are no set Methods yet as these aren't needed yet. I could change Rename to NameSet if you want.